// neo4j-driver-core/lib/internal/protocol-version.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtocolVersion = void 0;
var ProtocolVersion = /** @class */ (function () {
    function ProtocolVersion(major, minor) {
        this.major = major;
        this.minor = minor;
    }
    ProtocolVersion.prototype.getMajor = function () {
        return this.major;
    };
    ProtocolVersion.prototype.getMinor = function () {
        return this.major;
    };
    ProtocolVersion.prototype.isLessThan = function (other) {
        if (this.major < other.major) {
            return true;
        }
        else if (this.major === other.major && this.minor < other.minor) {
            return true;
        }
        return false;
    };
    ProtocolVersion.prototype.isGreaterThan = function (other) {
        if (this.major > other.major) {
            return true;
        }
        else if (this.major === other.major && this.minor > other.minor) {
            return true;
        }
        return false;
    };
    ProtocolVersion.prototype.isGreaterOrEqualTo = function (other) {
        return !this.isLessThan(other);
    };
    ProtocolVersion.prototype.isLessOrEqualTo = function (other) {
        return !this.isGreaterThan(other);
    };
    ProtocolVersion.prototype.isEqualTo = function (other) {
        return this.major === other.major && this.minor === other.minor;
    };
    ProtocolVersion.prototype.toString = function () {
        return this.major.toString() + '.' + this.minor.toString();
    };
    return ProtocolVersion;
}());
exports.ProtocolVersion = ProtocolVersion;
